packages/next/build/swc/src/next_dynamic.rs

```diff
@@ -0,0 +1,174 @@
+use swc_atoms::{js_word, JsWord};
+use swc_common::{FileName, DUMMY_SP};
+use swc_ecmascript::ast::{
+  ArrayLit, ArrowExpr, BinExpr, BinaryOp, BlockStmtOrExpr, CallExpr, Expr, ExprOrSpread,
+  ExprOrSuper, Ident, ImportDecl, ImportSpecifier, KeyValueProp, Lit, MemberExpr, ObjectLit, Prop,
+  PropName, PropOrSpread, Str, StrKind,
+};
+use swc_ecmascript::visit::Fold;
+
+pub fn next_dynamic(filename: FileName) -> impl Fold {
+  NextDynamicPatcher {
+    filename,
+    dynamic_binding: js_word!(""),
+  }
+}
+
+#[derive(Debug)]
+struct NextDynamicPatcher {
+  filename: FileName,
+  dynamic_binding: JsWord,
```

Comment:
It's better to use `Id` from `swc_ecmascript::utils`.
See: https://docs.rs/swc_ecma_utils/0.41.0/swc_ecma_utils/ident/type.Id.html

SWC handles scoping via metadata in `span.ctxt` of `Ident` and `Id` consists of `(Ident.sym, Ident.span.ctxt)`.
Such metadata is added by [`resolver`](https://docs.rs/swc_ecma_transforms/0.63.0/swc_ecma_transforms/resolver/fn.resolver_with_mark.html) and in future, custom passes invoked after applying resolver, so that custom passes can use such metadata for distinguishing identifiers.

Reply:
It's like 

```ts
var a;
{
  let a;
  use(a);
}
use(a);
```

as

```ts
var a#1;
{
  let a#2;
  use(a#2);
}
use(a#1);
```

`#1` and `#2` are `Ident.span.ctxt` part.